docs: oel migrate and upgrade guide (#1804)
ory/docs: docs/self-hosted/oel/oauth2/upgrade.mdx

    enabled: true
# ...
```

### 2. Update the image tag

```
image:
  repository: europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel
  tag: <replace-with-current-image-tag>
# ...
```

### 3. Apply / install the helm chart

You can now apply the upgrade:

```bash
helm upgrade ory-oel-hydra ory/hydra --namespace ory -f values.yaml
```

### 4. Wait for the upgrade to propagate

Once the init container is done and the main container is running, the upgrade is complete.

## Upgrade steps without Helm Charts

### 1. Check Release Notes

Before upgrading, review the [changelog](changelog.mdx) for the new version to understand the changes, new features, and any
deprecations.

### 2. Backup Your Data

Ensure you have a complete backup of your database and configuration files. This step is crucial for restoring your system in case
of an upgrade failure.

### 3. Pull the Latest Docker Image

Pull the latest Ory Hydra OEL Docker image from the Ory Enterprise Docker Registry:

```bash
docker pull europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag>
```

### 5. Apply SQL Migrations

Before deploying the service, you need to apply SQL migrations:

```bash

> docker run -e DSN=cockroach://... europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- migrate sql -e -f /path/to/config.yaml

Reviewer: I would just mention $DSN  like you did  in OS to OEL migration as this would imply crdb (which is not described as prerequisite)

> ```bash
> docker run -e DSN=cockroach://... europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- migrate sql -e -f /path/to/config.yaml
> ```
> 

Reviewer: ````suggestion
```bash
docker run -e DSN=your_database_connection_string europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- migrate sql -e -f /path/to/config.yaml
```

Replace `your_database_connection_string` with your actual database connection string.
````

### 5. Stop the Service

Once done, you can stop the processes running the old service.

### 6. Start the Service

Now you will be able to start the service:

```bash

> docker run -e DSN=cockroach://... europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- serve all -f /path/to/config.yaml

Reviewer: same with DSN and for consistency `-p 4444:4444 -p 4445:4445`

Author: 👍

> docker run -e DSN=cockroach://... europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- serve all -f /path/to/config.yaml

Author: ```suggestion
docker run -p 4444:4444 -p 4445:4445 -e DSN=cockroach://... europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- serve all -f /path/to/config.yaml
```

> docker run -p 4444:4444 -p 4445:4445 -e DSN=cockroach://... europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- serve all -f /path/to/config.yaml

Reviewer: ````suggestion
docker run -p 4444:4444 -p 4445:4445 -e DSN=your_database_connection_string europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<new-version-tag> -- serve all -f /path/to/config.yaml

Replace `your_database_connection_string` with your actual database connection string.
````
